### PR TITLE
Implement realnum::rand

### LINKED ISSUE #214: Randomness: implementation and use getrandom for Wasm target
### I'm submitting a 
- [x] feature request.

### Current Behaviour:
`RealNumber::rand` is not implemented. `Floatnumber::rand` use an external crate that cannot work on Wasm. 

the use of random in `fn rand()` is ambiguous and may fail in some targets. Some decisions have to be taken about the usage of `rand::Rng` also. I suggest going for using `getrandom` as it applies to the most [targets](https://docs.rs/getrandom/latest/getrandom/).

cc: @morenol

### PR DESCRIPTION
**Pull request by Mec-iS**
Implement rand. Use the new derive [#default]

Fixes #214 

### Checklist
- [x] My branch is up-to-date with development branch.
- [x] Everything works and tested on latest stable Rust.
- [x] Coverage and Linting have been applied 

### Current behaviour
`realnum::rand` returns `1.0`
### New expected behaviour
random number is generated

### CODE REVIEW

File: src/numbers/realnum.rs [hunk 1]
```diff
@@ -63,8 +66,12 @@ impl RealNumber for f64 {
     }
 
     fn rand() -> f64 {
-        // TODO: to be implemented, see issue smartcore#214
-        1.0
+        let mut small_rng = SmallRng::from_seed(SEEDS);
```
Comment (morenol) [inline src/numbers/realnum.rs]: could you call https://github.com/smartcorelib/smartcore/blob/development/src/rand_custom.rs here? so it can compile for wasm targets? both browser and non-browser tagets

**Comment by Mec-iS** [comment]
thanks @morenol
I need a `0.3.1` release asap! 👍🏼

File: src/numbers/realnum.rs [hunk 2]
```diff
@@ -66,7 +67,7 @@ impl RealNumber for f64 {
     }
 
     fn rand() -> f64 {
-        let mut small_rng = SmallRng::from_seed(SEEDS);
+        let mut small_rng = get_rng_impl(Some(100101010));
```
Comment (morenol) [inline src/numbers/realnum.rs]: I think that if we pass `None` here, we will get a random seed for std targets. We could change `get_rng_impl` to get a seed from current time for other targets maybe?

**Comment by morenol** [comment]
> thanks @morenol I need a `0.3.1` release asap! 👍🏼

Can you bump the version on this PR?

File: src/cluster/dbscan.rs [hunk 3]
```diff
@@ -18,7 +18,7 @@
 //!
 //! Example:
 //!
-//! ```
+//! ```ignore
```
Comment (morenol) [inline src/cluster/dbscan.rs:21]: Why is this needed?

**Review by morenol — DISMISSED** [review]
LGTM but not sure why `ignore` was added to one of the doc tests

**Comment by Mec-iS** [comment]
> LGTM but not sure why `ignore` was added to one of the doc tests

that test uses `datasets` feature so it fails, it is replicated among the `tests` with the right flag.

I may have more changes in the next hours, I am adding anti-panicking checks to `fit` methods.

**Comment by morenol** [comment]
> 
> I may have more changes in the next hours, I am adding anti-panicking checks to `fit` methods.

I like that idea!

**Comment by morenol** [comment]
It seems that there is conflicts, it says to me that there are 83 commits on this PR, could you rebase?

**Comment by Mec-iS** [comment]
>It seems that there is conflicts, it says to me that there are 83 commits on this PR, could you rebase?

that is because I pushed the wrong branch at the creation of the PR, then I corrected it.
Check in "Files Changed" that the changes are only the latest ones.

EDIT: your local branch may be missing the "Update README.md" I pushed before this PR.
